src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -351,23 +343,72 @@ private bool HasJobAttribute(CustomAttributeData attributeData)
             };
         }
 
-        private class ListenerFactory : IListenerFactory
+        internal class ListenerFactory : IListenerFactory
         {
             private readonly FunctionDescriptor _descriptor;
             private readonly ITriggeredFunctionExecutor _executor;
             private readonly ITriggerBinding _binding;
+            private readonly TraceWriter _trace;
 
-            public ListenerFactory(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, ITriggerBinding binding)
+            public ListenerFactory(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, ITriggerBinding binding, TraceWriter trace)
             {
                 _descriptor = descriptor;
                 _executor = executor;
                 _binding = binding;
+                _trace = trace;
             }
 
-            public Task<IListener> CreateAsync(CancellationToken cancellationToken)
+            public async Task<IListener> CreateAsync(CancellationToken cancellationToken)
             {
                 ListenerFactoryContext context = new ListenerFactoryContext(_descriptor, _executor, cancellationToken);
-                return _binding.CreateListenerAsync(context);
+                IListener listener = await _binding.CreateListenerAsync(context);
+                return new TriggerListener(listener, _descriptor, _trace);
+            }
+
+            private class TriggerListener : IListener
```

Comment:
I'm ensuring that only listeners associated with function triggers could be handled, as other listeners might have unintended side effects. Also, we only have information for the function metadata (name, etc) for these listeners, so we can throw errors that will be routed to the errant function.

Also, it ensures that I don't need to do any hacking with CompositeListener to pass aggregate exceptions.

I'll do a pass over the other listeners to determine if they need to be added.